### PR TITLE
More gun variants, part the second

### PR DESCRIPTION
**Pull request by DoctorBoomstick**
#### Summary
None

#### Purpose of change
Continues work started in PR #61598, flushing out the current in game guns which make use of the variants system.

#### Describe the solution
I’ve worked through some of the more commonly encountered and simpler in game 9x19mm guns and added variants for them. In no particular order, the guns I have modified are
1.	The Calico M960.
2.	The Berretta M9A1 and Berretta 90-two.
3.	The Cx4 Storm.
4.	 The Glock 17.
5.	The Glock 18C.
6.	The Glock 19.
7.	The M17.
8.	The M18.
In doing so, I have also added variants for the three compatible M17 and M18 magazines, whilst also merging the Berretta M9A1 and 90-two under a single variant due to their similar stats.
Apart for these changes, I have also revised the variant shotgun names I created in PR #57140, taking into account suggestions and advice gained from a lengthy discussion in the discord chat. As a miner aside, I have also altered some of the grammar used in the variant descriptions, with the goal of making them more readable and cutting down on giant, never ending sentences.

#### Describe alternatives you've considered
I considered finishing work on adding variants to all the 9x19mm guns before putting out this PR, but decided against it as I was feeling the impending doom of burn out on the horizon. Regardless, I also wanted to put the additions out for inspection.

#### Testing
The game loaded up with no errors and or noticeable issues, and the modified guns displaied the added variants.

#### Additional context
consult PR #61598 for the first part of this expansion to the existing gun variants.

### CODE REVIEW

**Review by GuardianDll — APPROVED** [review]
Good job!

File: data/json/items/gun/9mm.json [hunk 1]
```diff
@@ -1,43 +1,21 @@
 [
-  {
-    "id": "90two",
```
Comment (anothersimulacrum) [inline data/json/items/gun/9mm.json:3]: This needs a migration, see https://github.com/CleverRaven/Cataclysm-DDA/blob/a2e692099ce1d4f3c21354977b2d02e66c315fa8/data/json/items/migration.json#L1156-L1161 for an example.

**Review by anothersimulacrum — APPROVED** [review]
Just need the migration, then this looks great!

Merged guns:
| id | weight | volume | length | ammotype | modes | magazines | dispersion | recoil | damage | 
| --- | --- | --- | --- | --- | --- | --- | --- | --- | --- | 
| 90two | 921 | 554 | 250 | 9mm | 1 | ['m9mag', 'm9mag_10rd', 'm9mag_17rd', 'm9mag_18rd', 'm9mag_20rd', 'm9bigmag', 'm9mag_32rd', 'm9mag_35rd'] | 480 | 0 | 0 | 
| m9 | 961 | 570 | 249 | 9mm | 1 | ['m9mag', 'm9mag_10rd', 'm9mag_17rd', 'm9mag_18rd', 'm9mag_20rd', 'm9bigmag', 'm9mag_32rd', 'm9mag_35rd'] | 480 | 0 | 0 |

File: data/json/items/gun/9mm.json [hunk 2]
```diff
@@ -944,8 +962,17 @@
...
-    "name": { "str": "Glock 18C" },
-    "description": "A selective fire variation on the Glock 17, originally designed for Austria's EKO Cobra unit.  It has compensator cuts along its barrel to make recoil more manageable.",
+    "name": { "str": "Glock machine pistol" },
+    "description": "A military, fully automatic capable version of the popular 9x19mm handguns produced by the Glock arms company, this machine pistol makes use of Glock magazines.  The weapon has had its barrel modified to lesson recoil, a useful feature considering its high rate of fire and lack of a default buttstock.",
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "glock_18c",
+        "name": { "str": "Glock 18C" },
+        "description": "A selective fire variation on the Glock 17, originally designed for Austria's EKO Cobra unit.  It has compensator cuts along its barrel to make recoil more manageable.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/9mm.json:972]:
```suggestion
        "description": "A selective-fire variation on the Glock 17, originally designed for Austria's EKO Cobra unit.  It has compensator cuts along its barrel to make recoil more manageable.",
```

File: data/json/items/gun/9mm.json [hunk 3]
```diff
@@ -995,8 +1022,17 @@
     "copy-from": "pistol_base",
     "looks_like": "glock_17",
     "type": "GUN",
-    "name": { "str": "M17" },
-    "description": "The M17 is a semi-automatic, short recoil operated pistol derived from the SIG Sauer P320.",
+    "name": { "str": "service handgun" },
+    "description": "The general issue military handgun of the United States army before the Cataclysm, this is a sizable pistol that fires 9x19mm cartridges.  Based off successful civilian designs and capable of taking magazines of various round capacities, the gun features a plastic polymer frame which helps to reduce overall weight.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/9mm.json:1026]:
```suggestion
    "description": "The general-issue military handgun of the United States army before the Cataclysm, this is a sizable pistol that fires 9x19mm cartridges.  Based off successful civilian designs and capable of taking magazines of various round capacities, the gun features a plastic polymer frame which helps to reduce overall weight.",
```

File: data/json/items/gun/9mm.json [hunk 4]
```diff
@@ -995,8 +1022,17 @@
...
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "m17",
+        "name": { "str": "M17" },
+        "description": "The M17 is a semi-automatic, short recoil operated pistol derived from the SIG Sauer P320.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/9mm.json:1032]: I would have kept the military usage in this variant desc., like what you put for the M18.

File: data/json/items/gun/9mm.json [hunk 5]
```diff
@@ -1023,8 +1059,17 @@
     "id": "m18",
     "copy-from": "m17",
     "type": "GUN",
-    "name": { "str": "M18" },
-    "description": "The M18 is a compact, semi-automatic, short recoil operated pistol derived from the SIG Sauer P320 Carry, used by US Armed Forces.  Practically the same as its big brother, the M17, in every respect, the M18 features a moderately shorter slide and barrel which terminates at the end of the frame rather than extending partway beyond it like the M17.",
+    "name": { "str": "compact service handgun" },
+    "description": "A more petite version of the United States army’s general issue sidearm, this is a modestly sized pistol firing 9x19mm rounds.  Making use of the same magazines as its larger brother, the weapon features a shorter barrel and frame which helps to reduce its overall profile and make it more concealable as a result.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/9mm.json:1063]:
```suggestion
    "name": { "str": "compact service handgun" },
    "description": "A more petite version of the United States army’s general issue sidearm, this is a modestly-sized pistol firing 9x19mm rounds.  Making use of the same magazines as its larger brother, the weapon features a shorter barrel and frame which helps to reduce its overall profile and make it more concealable as a result.",
```

File: data/json/items/gun/shot.json [hunk 6]
```diff
@@ -193,9 +193,9 @@
...
     "looks_like": "remington_870",
     "type": "GUN",
-    "name": { "str": "12 gauge combat shotgun" },
-    "//": "Not exactly a combat shotgun, but I was hard-pressed to think of a name which wasn’t just, “12 gauge shotgun.” in its proper descriptions its referenced as often being used by militaries, there for I’m calling this a combat shotgun.",
-    "description": "The faithful friend of law-enforcement officers, military personnel, and zombie-hunting action heroes alike, this quaint firearm with its over 20 inch long barrel, 6-round magazine tube, and iconic pump-action design is as simple as it is effective.  One of the most popular firearm designs in all of history, this weapon can be reliably counted upon to do the job whether you’re simply looking to hunt, or to, with the right loads, turn the heads of the living dead into a chunky zombie puree.",
+    "name": { "str": "6-round combat shotgun" },
+    "//": "Not exactly a combat shotgun, but I was hard-pressed to think of a name which wasn’t just, “6-round shotgun.” in its proper descriptions its referenced as often being used by militaries, there for I’m calling this a combat shotgun.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/shot.json:197]:
```suggestion
    "//": "Not exactly a combat shotgun, but I was hard-pressed to think of a name which wasn’t just, “6-round shotgun.” In its proper descriptions it's referenced as often being used by militaries, therefore I’m calling this a combat shotgun.",
```

File: data/json/items/gun/shot.json [hunk 7]
```diff
@@ -231,8 +231,8 @@
     "id": "mossberg_590",
     "copy-from": "mossberg_500",
     "type": "GUN",
-    "name": { "str": "pump-action assault shotgun" },
-    "description": "A military oriented spin on the classic pump-action shotgun recipe, this firearm features an extended magazine tube capable of fielding 9 rounds of 12 gauge shotshells.  Sporting a heavier barrel and a bayonet lug, it makes for a highly intimidating weapon, as well as a satisfying answer to most of the nasties you’re liable to find in the post-apocalypse.",
+    "name": { "str": "9-round combat shotgun" },
+    "description": "A military oriented spin on the classic pump-action shotgun recipe, this firearm features an extended magazine tube capable of fielding 9 rounds of 12 gauge shotshells.  Sporting a heavier barrel and a bayonet lug, it makes for a highly intimidating weapon, as well as a satisfying answer to most of the nasties you're liable to find in the post-apocalypse.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/shot.json:235]:
```suggestion
    "description": "A military oriented spin on the classic pump-action shotgun recipe, this firearm features an extended magazine tube capable of fielding nine rounds of 12 gauge shotshells.  Sporting a heavier barrel and a bayonet lug, it makes for a highly intimidating weapon, as well as a satisfying answer to most of the nasties you're liable to find in the post-apocalypse.",
```

File: data/json/items/gun/shot.json [hunk 8]
```diff
@@ -677,8 +677,8 @@
...
+    "name": { "str": "auto-loading bullpup shotgun" },
+    "description": "Looking more like some form of sci-fi prop gun than an actual firearm, this weapon uses a system of triple magazines to field 15 12-gauge shotgun shells within a comparatively compact design.  Each tube has to be loaded separately, but this offers the option of loading different ammunition for different situations.  As the weapon’s trigger assembly is situated forwards of the chamber and magazines in what is termed as a 'bullpup' design, the firearm’s notably more compact than its peers despite having a higher capacity when compared against more classically designed shotguns.",
```
Comment (TheShadowFerret) [inline data/json/items/gun/shot.json:681]: "Triple magazine" says "one conjoined object made of three individual magazines" to me

**Review by TheShadowFerret — CHANGES_REQUESTED** [review]

File: data/json/items/gun/9mm.json [hunk 9]
```diff
@@ -1,43 +1,21 @@
...
+    "name": { "str": "Calico automatic carbine" },
+    "description": "A compact submachine gun designed evidently under the 'spray and pray' mentality, this Calico automatic carbine is not notably powerful, or dependable, or even particularly accurate to shoot.  Intended for military and law enforcement use, it never saw any noteworthy service due to its poor reliability and overall performance, and would be an unremarkable weapon in all respects, save for its enormous magazine capacity.  Making use of uniquely designed 50 or 100 round circular magazines, the Calico might not be a particularly good gun, but will throw an overwhelming amount of led down range for a long time providing that it doesn’t choke, jam, and die.",
+    "variant_type": "gun",
+    "variants": [
+      {
+        "id": "calico",
+        "name": { "str_sp": "Calico M960" },
+        "//": "Automatic variant.  Cyclic fire rate: 750 rpm",
+        "description": "The Calico M960 is an automatic carbine with a unique circular magazine that allows for high capacities and reduced recoil.",
```
[the remark this answers is not in the document]
Comment (DoctorBoomstick) [reply data/json/items/gun/9mm.json:15]: Would it be best if I took a buzzsaw to the variant descriptions and trimmed off a few sentences to make them more compact, or would switching them and the regular descriptions be a better answer? I don’t want to tread on anyone’s toes here, but I also do like me my bloated, overcooked and pointlessly rambling descriptions.

**Review by TheShadowFerret — APPROVED** [review]

**Pull request merged**